### ISSUE TITLE
viewDidAppear called twice during state restoration

### ISSUE THREAD

**Issue by AndriyGo** [mutualmobile/MMDrawerController#261]
In the center view controller (and, probably, others as well) viewDidAppear method is called twice when restoring state.

1) The first call is triggered by 

[self.centerViewController beginAppearanceTransition:YES animated:NO];
[self.centerViewController endAppearanceTransition];

in the (setCenterViewController: animated:) method of MMDrawerController. 

2) Second time it is called from viewWillAppear/viewDidAppear methods of MMDrawerController due to

a) [self.centerViewController beginAppearanceTransition:YES animated:animated]; in viewWillAppear
b) [self.centerViewController endAppearanceTransition]; in viewDidAppear

If the state is not getting restored, centre view controller's viewDidAppear is not called due to (1) and is only called due to (2).

I have tested and this is also the case with KitchenSink. I'd be happy to find a solution for this and issue a pull request, but I am stuck trying to understand why (1) calls viewDidAppear ONLY when restoring state.

**Comment by gotitplatform** [comment]
I have a MMDrawerController that is pushed onto a UINavigationController stack not animated and I get multiple calls to viewWillAppear and viewDidAppear

UPDATE:
I think I found the issue. I had a UINavigationController displayed behind a modal UIViewController. I was calling push (not animated) after I made an animated call to dismiss the modal. Moving the push (not animated) before calling dismiss (animated) on the modal achieved the results I was looking for and makes sense as to what was happening.

**TLDR:** Nevermind...
